src/main/java/com/minecolonies/coremod/entity/ai/citizen/guard/AbstractEntityAIGuard.java

```diff
@@ -1146,4 +1147,25 @@ public boolean canBeInterrupted()
         }
         return super.canBeInterrupted();
     }
+
+    /**
+     * Method which determines whether to apply attack actions.
+     *
+     * @return the boolean of whether or not to apply attack actions.
+     */
+    public boolean handleAttackExceptions(final Entity targetEntity)
+    {
+        if (targetEntity instanceof PlayerEntity)
+        {
+            if( worker.getCitizenColonyHandler().getColony() != null && worker.getCitizenColonyHandler().getColony().getPermissions().isColonyMember((PlayerEntity) targetEntity))
```

Comment:
maybe the check should here check for the guard attack permission

Reply:
So you want to use this method, but change the method to work differently?